allennlp/training/callbacks/wandb.py

```diff
@@ -134,12 +127,31 @@ def _log(
             dict_to_log = {f"{log_prefix}/{k}": v for k, v in dict_to_log.items()}
         if epoch is not None:
             dict_to_log["epoch"] = epoch
-        self.wandb.log(dict_to_log, step=self.trainer._batch_num_total)  # type: ignore[union-attr]
+        self.wandb.log(dict_to_log, step=self.trainer._batch_num_total)  # type: ignore
 
     @overrides
     def on_start(
         self, trainer: "GradientDescentTrainer", is_primary: bool = True, **kwargs
     ) -> None:
         super().on_start(trainer, is_primary=is_primary, **kwargs)
+
+        if not is_primary:
+            return None
+
+        import wandb
+
+        self.wandb = wandb
```

Comment:
There is no `wandb` object? `wandb` is always global? What if two systems want to use it at the same time?

Reply:
Importing `wandb` may have side effects since at some point it spawns its own background worker(s).

Reply:
That is some unfortunate API design.